packages/ui/src/components/Visualization/Canvas/FormV2/fields/EnumField.tsx

```diff
@@ -0,0 +1,79 @@
+import { FormGroup, FormGroupLabelHelp, Popover } from '@patternfly/react-core';
+import { FunctionComponent, useCallback, useContext, useMemo } from 'react';
+import { isDefined } from '../../../../../utils';
+import { useFieldValue } from '../hooks/field-value';
+import { SchemaContext } from '../providers/SchemaProvider';
+import { FieldProps } from '../typings';
+import { Typeahead } from '../../../../typeahead/Typeahead';
+import { TypeaheadItem } from '../../../../typeahead/Typeahead.types';
+
+export const EnumField: FunctionComponent<FieldProps> = ({ propName, required }) => {
+  const { schema } = useContext(SchemaContext);
+  const { value = '', onChange } = useFieldValue<string | undefined>(propName);
+
+  const items: TypeaheadItem<string>[] = useMemo(
+    () =>
+      schema?.enum?.map((item) => ({
+        name: String(item),
+        description: '',
+        value: String(item),
+      })) || [],
+    [schema],
+  );
+
+  const selectedItem = useMemo(() => {
+    if (!value) {
+      return undefined;
+    }
+    return items.find((item) => item.name === value);
+  }, [items, value]);
+
+  const onItemChange = useCallback(
+    (item?: TypeaheadItem<string>) => {
+      if (!isDefined(item)) {
+        onChange(undefined);
+        return;
+      }
+      onChange(item.name);
```

Comment:
We could simplify it to:

```suggestion
      onChange(item?.name);
```